BusinessTampereTrafficMonitoring/stream_reader/stream_reader.py

```diff
@@ -0,0 +1,102 @@
+import logging
+import queue
+import time
+from collections import deque
+from threading import Thread
+
+import cv2
+import numpy as np
+
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.ERROR)
+
+formatter = logging.Formatter('%(asctime)s:%(name)s:%message)s')
+
+file_handler = logging.FileHandler('stream_reader-log')
+file_handler.setFormatter(formatter)
+
+logger.addHandler(file_handler)
+
+
+class StreamReader:
+    def __init__(self, video_location):
+
+        self.buffer = queue.Queue()
+        self.cache = dict()
+
+        self.timestamps = deque([])
+        self.t_latest_frame = 0
+        self.video_location = video_location
+
+        self.cap = cv2.VideoCapture(self.video_location, apiPreference=cv2.CAP_FFMPEG)
+        if not self.cap.isOpened():
+            print('Cannot open stream')
+            exit(-1)
```

Comment:
This was also here before these changes but it would probably make more sense to raise an exception rather than print and quit.